packages/iov-ethereum/src/derivation.ts

```diff
@@ -0,0 +1,23 @@
+import { Address } from "@iov/bcp-types";
+import { Keccak256 } from "@iov/crypto";
+import { Encoding } from "@iov/encoding";
+
+const { toAscii, toHex } = Encoding;
+
+export function isValidAddress(address: string): boolean {
+  if (address.match(/^0x[a-fA-F0-9]+$/) && address.length === 42) {
+    // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md
+    const addressLower = address.toLowerCase().replace("0x", "") as Address;
+    const addressHash = toHex(new Keccak256(toAscii(addressLower)).digest());
+    for (let i = 0; i < 40; i++) {
+      if (
+        (parseInt(addressHash[i], 16) > 7 && addressLower[i].toUpperCase() !== address[i + 2]) ||
+        (parseInt(addressHash[i], 16) <= 7 && addressLower[i] !== address[i + 2])
+      ) {
+        return false;
+      }
+    }
+    return true;
+  }
+  throw new Error("Invalid ethereum address");
```

Comment:
Just return `false` in this case, no exception. `isValidAddress` is not for checksum matching but also returns false when you enter a Lisk address in an Ethereum transaction.